inc/saitam.h

```diff
@@ -868,29 +1012,2440 @@ typedef sai_status_t (*sai_get_tam_transporter_attribute_fn)(
+    SAI_TAM_EVENT_THRESHOLD_ATTR_CUSTOM_RANGE_END
+
+} sai_tam_event_threshold_attr_t;
+
+/**
+ * @brief Create and return a threshold object
+ *
+ * @param[out] tam_event_threshold_id Event Threshold object
+ * @param[in] switch_id Switch object id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_tam_event_threshold_fn)(
+        _Out_ sai_object_id_t *tam_event_threshold_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Deletes a specified threshold object
+ *
+ * @param[in] tam_event_threshold_id Event Threshold object
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_tam_event_threshold_fn)(
+        _In_ sai_object_id_t tam_event_threshold_id);
+
+/**
+ * @brief Get values for specified threshold object attributes
+ *
+ * @param[in] tam_event_threshold_id Event Threshold object
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_tam_event_threshold_attribute_fn)(
+        _In_ sai_object_id_t tam_event_threshold_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Set value for a specified threshold object attribute
+ *
+ * @param[in] tam_event_threshold_id Event Threshold object
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_tam_event_threshold_attribute_fn)(
+        _In_ sai_object_id_t tam_event_threshold_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief TAM telemetry types supported
+ */
+typedef enum _sai_tam_telemetry_type_t
+{
+    /**
+     * @brief Networking element TAM
+     */
+    SAI_TAM_TELEMETRY_TYPE_NE,
+
+    /**
+     * @brief Switch silicon TAM
+     */
+    SAI_TAM_TELEMETRY_TYPE_SWITCH,
+
+    /**
+     * @brief Fabric TAM
+     */
+    SAI_TAM_TELEMETRY_TYPE_FABRIC,
+
+    /**
+     * @brief Flow TAM
+     */
+    SAI_TAM_TELEMETRY_TYPE_FLOW,
+
+    /**
+     * @brief Networking element TAM
+     */
+    SAI_TAM_TELEMETRY_TYPE_INT
+
+} sai_tam_telemetry_type_t;
+
+/**
+ * @brief INT telemetry attributes
+ */
+typedef enum _sai_tam_tel_int_attr_t
+{
+
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_TEL_INT_ATTR_START,
+
+    /**
+     * @brief INT template Id
+     *
+     * Template id indicates the supported metadata header.
+     * Device can support one or more template IDs
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_TEL_INT_ATTR_TEMPLATE_ID = SAI_TAM_TEL_INT_ATTR_START,
```

Comment:
Where is the "template" defined. Not found in this spec

Reply:
Template Id as specified in https://tools.ietf.org/html/draft-kumar-ifa-00